### PR TITLE
Build Finschia layer-1 binary

### PR DESCRIPTION
**Pull request by torao**
## Description

This PR adds Rollup, DA, and Settlement modules to the `hub_for_ramus` branch to build the L1 version of finschia blockchain.

Note that it is intended for PoC purposes. IBC and WASM functionality has been removed for avoid modification affecting many repositories.

## How has this been tested?

pass `make test`

### CODE REVIEW

File: app/app.go [hunk 1]
```diff
@@ -170,10 +150,9 @@ var (
 		vesting.AppModuleBasic{},
 		tokenmodule.AppModuleBasic{},
 		collectionmodule.AppModuleBasic{},
-		ibc.AppModuleBasic{},
-		transfer.AppModuleBasic{},
-		ica.AppModuleBasic{},
-		wasmplus.AppModuleBasic{},
+		rollup.AppModuleBasic{},
+		// ordamodule.AppModuleBasic{},
```
Comment (Mdaiki0730) [inline app/app.go:154]: Why comment out da module?

File: app/app.go [hunk 2]
```diff
@@ -232,20 +209,16 @@ type LinkApp struct { // nolint: golint
...
-	WasmKeeper       wasmpluskeeper.Keeper
+	RollupKeeper     rollupkeeper.Keeper
+	Ordakeeper       ordakeeper.Keeper
+	SettlementKeeper settlementkeeper.Keeper
 
-	ScopedIBCKeeper      capabilitykeeper.ScopedKeeper
-	ScopedTransferKeeper capabilitykeeper.ScopedKeeper
-	ScopedICAHostKeeper  capabilitykeeper.ScopedKeeper
-	ScopedWasmKeeper     capabilitykeeper.ScopedKeeper
+	ScopedICAHostKeeper capabilitykeeper.ScopedKeeper
```
Comment (Mdaiki0730) [inline app/app.go:221]: Since the other descriptions of ICA have been removed, would you like to do the same?

File: app/app.go [hunk 3]
```diff
@@ -513,15 +411,14 @@ func NewLinkApp(
...
 		evidence.NewAppModule(app.EvidenceKeeper),
 		params.NewAppModule(app.ParamsKeeper),
 		tokenmodule.NewAppModule(appCodec, app.TokenKeeper),
 		collectionmodule.NewAppModule(appCodec, app.CollectionKeeper),
 		authzmodule.NewAppModule(appCodec, app.AuthzKeeper, app.AccountKeeper, app.BankKeeper, app.interfaceRegistry),
-		ibc.NewAppModule(app.IBCKeeper),
-		transferModule,
-		icaModule,
+		rollup.NewAppModule(appCodec, app.RollupKeeper, app.AccountKeeper, app.BankKeeper),
+		// ordamodule.NewAppModule(appCodec, app.Ordakeeper, app.AccountKeeper),
```
Comment (Mdaiki0730) [inline app/app.go:420]: Why comment out da module?

File: app/app.go [hunk 4]
```diff
@@ -870,10 +742,8 @@ func initParamsKeeper(appCodec codec.BinaryCodec, legacyAmino *codec.LegacyAmino
 	paramsKeeper.Subspace(govtypes.ModuleName).WithKeyTable(govtypes.ParamKeyTable())
 	paramsKeeper.Subspace(crisistypes.ModuleName)
 	paramsKeeper.Subspace(foundation.ModuleName)
-	paramsKeeper.Subspace(ibctransfertypes.ModuleName)
-	paramsKeeper.Subspace(ibchost.ModuleName)
-	paramsKeeper.Subspace(icahosttypes.SubModuleName)
-	paramsKeeper.Subspace(wasmplustypes.ModuleName)
+	paramsKeeper.Subspace(settlementtypes.ModuleName)
+	paramsKeeper.Subspace(rolluptypes.ModuleName)
```
Comment (Mdaiki0730) [inline app/app.go:746]:
```suggestion
	paramsKeeper.Subspace(settlementtypes.ModuleName)
	paramsKeeper.Subspace(ordatypes.ModuleName)
	paramsKeeper.Subspace(rolluptypes.ModuleName)
```

File: app/app.go (hunk 1 above)
Comment (torao) [reply app/app.go]: This procedure is a direct migration of the `finschia-sdk` code. It's true that no `x/or/da` modules have been added. In conjunction with the points made subsequently, this part of the code will be reinstated.

File: app/app.go (hunk 3 above)
Comment (torao) [reply app/app.go]: DITTO.

File: app/app.go (hunk 4 above)
Comment (torao) [reply app/app.go]: DITTO.

File: app/app.go (hunk 2 above)
Comment (torao) [reply app/app.go]: Indeed, this field is already no longer used. I'll delete this.

File: app/app.go (hunk 1 above)
Comment (shiki-tak) [reply app/app.go]: If you want to remove the comment, please import the latest commit.
https://github.com/Finschia/finschia-sdk/commit/25043656ed1a6a4947b2c5c3804d724bbcfc43ec

**Review by Mdaiki0730 — APPROVED** [review]

**Review by shiki-tak — APPROVED** [review]
LGTM

**Pull request merged**